examples/bert/bert_finetune_glue.py

```diff
@@ -88,17 +88,21 @@ def load_data(task_name):
         task_name = "mnli"
         test_suffix = "_mismatched"
 
-    def to_tf_dataset(split):
-        # Format each sample as a tuple of string features and an int label.
-        features = tuple([split[f] for f in feature_names])
-        label = tf.cast(split["label"], tf.int32)
-        return tf.data.Dataset.from_tensor_slices((features, label))
-
-    data = datasets.load_dataset("glue", task_name)
-    data.set_format(type="tensorflow")
-    train_ds = to_tf_dataset(data["train"])
-    test_ds = to_tf_dataset(data["test" + test_suffix])
-    validation_ds = to_tf_dataset(data["validation" + test_suffix])
+    def preprocess_data(ds):
```

Comment:
what about something like `def split_features(x):` which does the inner lambda, and then you call three times (for each dataset

`train_ds = train_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)`

That would be the most idiomatic